content/en/docs/concepts/services-networking/service.md

```diff
@@ -150,90 +189,129 @@ Every node in a Kubernetes cluster runs a `kube-proxy`. `kube-proxy` is
 responsible for implementing a form of virtual IP for `Services` of type other
 than [`ExternalName`](#externalname).
 
-In Kubernetes v1.0, `Services` are a "layer 4" (TCP/UDP over IP) construct, the
-proxy was purely in userspace.  In Kubernetes v1.1, the `Ingress` API was added
-(beta) to represent "layer 7"(HTTP) services, iptables proxy was added too,
-and became the default operating mode since Kubernetes v1.2. In Kubernetes v1.8.0-beta.0,
-ipvs proxy was added.
+### Why not use round-robin DNS?
+
+A question that pops up every now and then is why Kubernetes has uses proxying
+to direct incoming traffic to backends. Would it be possible to configure
+DNS records that have multiple A values (or AAAA for IPv6) and rely on round-robin
+name resoltion?
 
-### Proxy-mode: userspace
+There are a few reasons for using proxying for Services:
+
+ * There is a long history of DNS libraries not respecting DNS TTLs and
+   caching the results of name lookups.
+ * Many apps do DNS lookups only once and cache the results indefinitely.
+ * Even if apps and libraries did proper re-resolution, the load of every
+   client re-resolving DNS over and over would be difficult to manage.
+
+### Version compatibility
+
+Services are a "layer 4" (TCP/UDP over IP) construct. Since Kubernetes v1.0
+you have been able to use the [userspace proxy mode](#proxy-mode-userspace).
+Kubernetes v1.1 added iptables mode proxying was added too, which became the
+default mode in Kubernetes v1.2.
+Kubernetes v1.8 added ipvs proxy mode.
+
+### User space proxy mode {#proxy-mode-userspace}
 
 In this mode, kube-proxy watches the Kubernetes master for the addition and
-removal of `Service` and `Endpoints` objects. For each `Service` it opens a
+removal of Service and Endpoint objects. For each Service it opens a
 port (randomly chosen) on the local node.  Any connections to this "proxy port"
-will be proxied to one of the `Service`'s backend `Pods` (as reported in
-`Endpoints`).  Which backend `Pod`  to use is decided based on the
-`SessionAffinity` of the `Service`.  Lastly, it installs iptables rules which
-capture traffic to the `Service`'s `clusterIP` (which is virtual) and `Port`
-and redirects that traffic to the proxy port which proxies the backend `Pod`.
-By default, the choice of backend is round robin.
+will be proxied to one of the Service's backend Pods (as reported via
+Endpoints).  Which backend Pod  to use is decided based on the
+`SessionAffinity` of the Service.
+
+Lastly, the user-space proxy installs iptables rules which capture traffic to
+the Service's `clusterIP` (which is virtual) and `port`. The rules
+redirect that traffic to the proxy port which proxies the backend Pod.
+
+By default, kube-proxy in userspace mode chooses a backend via a round-robin algorithm.
```

Comment:
nit: will be nice if we had some info about the round-robin algorithm so we can link it (i know this is for advanced users but some developers for examples will appreciate the details

Reply:
This PR is already huge! That feels like something to fix in a follow-up set of changes.

Reply:
okay, fair point